pkg/codegen/go/gen_program_expressions.go

```diff
@@ -312,9 +418,19 @@ func (g *generator) GenTupleConsExpression(w io.Writer, expr *model.TupleConsExp
 
 // GenTupleConsExpression generates code for a TupleConsExpression.
 func (g *generator) genTupleConsExpression(w io.Writer, expr *model.TupleConsExpression, destType model.Type) {
-	// TODO derive from ambient context
-	isInput := true
-	argType := g.argumentTypeName(expr, destType, isInput)
+	isInput := isInputty(destType) || containsInputs(expr)
+	argType := argumentTypeName(expr, destType, isInput)
+	if strings.HasSuffix(argType, "Array") {
+		isInput = true
+	}
```

Comment:
Shouldn't this be covered by `isInput`?

Reply:
Yup, this is leftover from the earliest implementation.